layouts/_default/single.html

```diff
@@ -6,7 +6,7 @@ <h1 class="post__title">{{ .Title }}</h1>
 			{{- with .Params.lead }}
 			<p class="post__lead">{{ . }}</p>
 			{{- end }}
-			{{- if or (not .Date.IsZero) .Params.categories }}<div class="post__meta meta">{{ partial "post_meta.html" . }}</div>{{ end }}
+			<div class="post__meta meta">{{ partial "post_meta.html" . }}</div>
```

Comment:
This make no sense, when users can drop their on meta data fields. I try to address the problem in style.css using `:not(:empty)` selectors.

Reply:
Suggested CSS doesn't solve main problems, it just hides them from users. Empty div remains in the code. I'll try to look for another solution (maybe `readDir` or `readFile`).

Reply:
How is an empty div problematic?

Accessing the file system directly adds one syscall per rendered post metadata and may cause a performance issue.

Another approach might be to execute the partial into a variable and check if it's empty. Don't know whether hugo supports this.

Reply:
> Another approach might be to execute the partial into a variable and check if it's empty. Don't know whether hugo supports this.

This worked and the code reads very nice.

Reply:
> How is an empty div problematic?

Technically this is not a problem, it's just extra bytes in some cases and not clean code.